src/utils.jl

```diff
@@ -0,0 +1,160 @@
+        new{S, T, N, 1, Array{T, 1}}(Array{T, 1}(undef, StaticArrays.tuple_prod(S)))
+    end
+    function SizedAbstractArray{S, T, N, N}(::UndefInitializer) where {S, T, N, TData<:AbstractArray}
+        new{S, T, N, N, Array{T, N}}(Array{T, N}(undef, size_to_tuple(S)...))
+    end
+end
+
+@inline SizedAbstractArray{S,T,N}(a::TData) where {S,T,N,M,TData<:AbstractArray{T,M}} = SizedAbstractArray{S,T,N,M,TData}(a)
+@inline SizedAbstractArray{S,T}(a::TData) where {S,T,M,TData<:AbstractArray{T,M}} = SizedAbstractArray{S,T,StaticArrays.tuple_length(S),M,TData}(a)
+@inline SizedAbstractArray{S}(a::TData) where {S,T,M,TData<:AbstractArray{T,M}} = SizedAbstractArray{S,T,StaticArrays.tuple_length(S),M,TData}(a)
+
+@inline SizedAbstractArray{S,T,N}(::UndefInitializer) where {S,T,N} = SizedAbstractArray{S,T,N,N}(undef)
+@inline SizedAbstractArray{S,T}(::UndefInitializer) where {S,T} = SizedAbstractArray{S,T,StaticArrays.tuple_length(S),StaticArrays.tuple_length(S)}(undef)
+@generated function (::Type{SizedAbstractArray{S,T,N,M,TData}})(x::NTuple{L,Any}) where {S,T,N,M,TData,L}
+    if L != StaticArrays.tuple_prod(S)
+        error("Dimension mismatch")
+    end
+    exprs = [:(a[$i] = x[$i]) for i = 1:L]
+    return quote
+        $(Expr(:meta, :inline))
+        a = SizedAbstractArray{S,T,N,M}(undef)
+        @inbounds $(Expr(:block, exprs...))
+        return a
+    end
+end
+
+@inline SizedAbstractArray{S,T,N}(x::Tuple) where {S,T,N} = SizedAbstractArray{S,T,N,N}(x)
+@inline SizedAbstractArray{S,T}(x::Tuple) where {S,T} = SizedAbstractArray{S,T,StaticArrays.tuple_length(S),StaticArrays.tuple_length(S)}(x)
+@inline SizedAbstractArray{S}(x::NTuple{L,T}) where {S,T,L} = SizedAbstractArray{S,T,StaticArrays.tuple_length(S),StaticArrays.tuple_length(S)}(x)
+
+# Overide some problematic default behaviour
+@inline convert(::Type{SA}, sa::SizedAbstractArray) where {SA<:SizedAbstractArray} = SA(sa.data)
+@inline convert(::Type{SA}, sa::SA) where {SA<:SizedAbstractArray} = sa
+
+# Back to Array (unfortunately need both convert and construct to overide other methods)
+import Base.Array
+@inline Array(sa::SizedAbstractArray) = Array(sa.data)
+@inline Array{T}(sa::SizedAbstractArray{S,T}) where {T,S} = Array(sa.data)
+@inline Array{T,N}(sa::SizedAbstractArray{S,T,N}) where {T,S,N} = Array(sa.data)
+
+@inline convert(::Type{AbstractArray}, sa::SizedAbstractArray) = sa.data
+@inline convert(::Type{AbstractArray{T}}, sa::SizedAbstractArray{S,T}) where {T,S} = sa.data
+@inline convert(::Type{AbstractArray{T,N}}, sa::SizedAbstractArray{S,T,N}) where {T,S,N} = sa.data
+
+Base.@propagate_inbounds getindex(a::SizedAbstractArray, i::Int) = getindex(a.data, i)
+Base.@propagate_inbounds setindex!(a::SizedAbstractArray, v, i::Int) = setindex!(a.data, v, i)
+
+SizedAbstractVector{S,T,M} = SizedAbstractArray{Tuple{S},T,1,M}
+@inline SizedAbstractVector{S}(a::AbstractArray{T,M}) where {S,T,M} = SizedAbstractArray{Tuple{S},T,1,M}(a)
```

Comment:
This doesn't seem to work for me
```julia
julia> x = randn(3)
3-element Array{Float64,1}:
  0.10460674079990906
  0.02796008883007997
 -0.2567903778403676 

julia> x2 = StaticVector{3}(x)
ERROR: DimensionMismatch("No precise constructor for StaticArray{Tuple{3},T,1} where T found. Length of input was 3.")
Stacktrace:
 [1] StaticArray{Tuple{3},T,1} where T(::Tuple{Tuple{Tuple{Tuple{Float64,Float64,Float64}}}}) at /Users/saxen/projects/StaticArrays.jl/src/convert.jl:1
 [2] Type at /Users/saxen/projects/StaticArrays.jl/src/convert.jl:4 [inlined] (repeats 3 times)
 [3] _convert at /Users/saxen/projects/StaticArrays.jl/src/convert.jl:29 [inlined]
 [4] convert at /Users/saxen/projects/StaticArrays.jl/src/convert.jl:26 [inlined]
 [5] StaticArray{Tuple{3},T,1} where T(::Array{Float64,1}) at /Users/saxen/projects/StaticArrays.jl/src/convert.jl:7

```

Reply:
`StaticVector{3}` is an abstract type from `StaticArrays` and I don't think it's supposed to work. Did you mean `x2 = Manifolds.SizedAbstractVector{3}(x)`? I didn't work but it should, so I've fixed that.

Reply:
Yes, sorry, that's what I meant.